File: internal/server/match/map.go
package match

import "github.com/lemunozm/ascii-arena/internal/pkg/spatial"
import "github.com/lemunozm/ascii-arena/internal/pkg/def"
import "hash/crc64"

type Map struct {
	width  int
	height int
	seed   string
	data   []def.Wall
}

func NewMap(width int, height int, seed string) *Map {
	m := &Map{
		width:  width,
		height: height,
		seed:   seed,
		data:   make([]def.Wall, width*height),
	}

	m.clean()
	m.buildBorder()
	m.generate()
	return m
}

func (m Map) GetWidth() int {
	return m.width
}

func (m Map) GetHeight() int {
	return m.height
}

func (m Map) GetSeed() string {
	return m.seed
}

func (m Map) GetData() []def.Wall {
	return m.data
}

func (m Map) GetBox(position spatial.Vector2) def.Wall { //To Vector
	return m.data[m.width*position.Y+position.X]
}

func (m *Map) SetBox(position spatial.Vector2, box def.Wall) {
	m.data[m.width*position.Y+position.X] = box
}

func (m *Map) clean() {
	for i := 0; i < m.width*m.height; i++ {
		m.data[i] = def.WALL_EMPTY
	}
}

func (m *Map) buildBorder() {
	for x := 0; x < m.width; x++ {
		m.SetBox(spatial.Vector2{x, 0}, def.WALL_BORDER) //comprobar la inicializacion rapida del vector
	}
	for x := 0; x < m.width; x++ {
		m.SetBox(spatial.Vector2{x, m.height - 1}, def.WALL_BORDER)
	}
	for y := 1; y < m.height-1; y++ {
		m.SetBox(spatial.Vector2{0, y}, def.WALL_BORDER)
	}
	for y := 1; y < m.height-1; y++ {
		m.SetBox(spatial.Vector2{m.width - 1, y}, def.WALL_BORDER)
	}
}

func (m *Map) generate() {
	crc64.Checksum([]byte(m.seed), crc64.MakeTable(crc64.ECMA))
	//TODO
}

File: src/com/asciiarena/lib/server/match/Map.java
package com.asciiarena.lib.server.match;

import com.asciiarena.lib.common.match.Wall;
import com.asciiarena.lib.common.util.Vector2;

public class Map
{
    private int width;
    private int height;
    private String seed;
    private Wall[][] grid;

    public Map(int width, int height, String seed)
    {
        this.width = width;
        this.height = height;
        this.seed = seed;
        this.grid = new Wall[width][height];

        clear();
        buildBorder();
        generate(seed.hashCode());
    }

    public void clear()
    {
        for(int y = 0; y < height; y++)
        {
            for(int x = 0; x < width; x++)
            {
                grid[x][y] = Wall.EMPTY;
            }
        }
    }

    private void buildBorder()
    {
        for(int x = 0; x < width; x++) 
        {
            grid[x][0] = Wall.BORDER;
        }

        for(int x = 0; x < width; x++) 
        {
            grid[x][height - 1] = Wall.BORDER;
        }

        for(int y = 1; y < height - 1; y++) 
        {
            grid[0][y] = Wall.BORDER;
        }

        for(int y = 1; y < height; y++) 
        {
            grid[width - 1][y] = Wall.BORDER;
        }
    }

    private void generate(int randomSeed)
    {
        //TODO
    }

    public void setPlace(Vector2 position, Wall wall)
    {
        grid[position.x][position.y] = wall;
    }

    public Wall getPlace(Vector2 position)
    {
        return grid[position.x][position.y];
    }

    public int getWidth()
    {
        return width;
    }

    public int getHeight()
    {
        return height;
    }

    public  String getSeed()
    {
        return seed;
    }

    public Wall[][] getGrid()
    {
        return grid;
    }
}
